code/statsd/src/main/scala/com/avast/datadog4s/statsd/StatsDMetricFactory.scala

```diff
@@ -61,13 +63,14 @@ class StatsDMetricFactory[F[_]: Sync](
       new GaugeDoubleImpl[F](statsDClient, extendPrefix(aspect), sampleRate.getOrElse(defaultSampleRate), defaultTags)
   }
 
-  override def timer(aspect: String, sampleRate: Option[Double] = None) =
+  override def timer(aspect: String, sampleRate: Option[Double] = None, timerMode: TimerMode = timerMode) =
```

Comment:
Wouldn't it be more efficient to have two different implementations of `Timer`? That way we only need to do pattern matching once on init time, and not every time the metric is updated.